[BLOCKED] System.CommandLine update

adamsitnik opened the pull request:
**Edit:** this PR is blocked until https://github.com/NuGet/NuGet.Client/pull/6236 gets merged (ETA is April).

Context: we are working on releasing a new public version of System.CommandLine. The only change we decided to revert was the adding of "Cli" prefix to all symbol types (because it would break every single user of the library).

adamsitnik committed:
adjust to System.CommandLine renaming

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/command-line-api build 20250122.1

Microsoft.SourceBuild.Intermediate.command-line-api , System.CommandLine , System.CommandLine.Rendering
 From Version 0.1.607102 -> To Version 0.1.607201

Evangelink reviewed (approved):
Approving for `dotnet test` and `dotnet vstest`

adamsitnik committed:
Merge remote-tracking branch 'upstream/main' into scl_prefix_update

# Conflicts:
#	src/Cli/dotnet/CommonOptions.cs
#	src/Cli/dotnet/commands/InstallingWorkloadCommand.cs
#	src/Cli/dotnet/commands/dotnet-run/RunCommandParser.cs
#	src/Cli/dotnet/commands/dotnet-test/Program.cs
#	src/Cli/dotnet/commands/dotnet-test/TestCommandParser.cs
#	src/Cli/dotnet/commands/dotnet-workload/search/WorkloadSearchVersionsCommandParser.cs

adamsitnik committed:
Merge branch 'main' into scl_prefix_update

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/sourcelink build 20250131.2

Microsoft.SourceBuild.Intermediate.sourcelink , Microsoft.Build.Tasks.Git , Microsoft.SourceLink.AzureRepos.Git , Microsoft.SourceLink.Bitbucket.Git , Microsoft.SourceLink.Common , Microsoft.SourceLink.GitHub , Microsoft.SourceLink.GitLab
 From Version 10.0.0-beta.25080.2 -> To Version 10.0.0-beta.25081.2

adamsitnik committed:
Merge remote-tracking branch 'upstream/main' into scl_prefix_update

# Conflicts:
#	eng/Versions.props

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/arcade build 20250203.6

Microsoft.SourceBuild.Intermediate.arcade , Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.SignTool , Microsoft.DotNet.XliffTasks , Microsoft.DotNet.XUnitExtensions
 From Version 10.0.0-beta.25080.7 -> To Version 10.0.0-beta.25103.6

adamsitnik committed:
Merge branch 'main' into scl_prefix_update
